RecoLocalCalo/EcalRecProducers/plugins/EcalUncalibRecHitPhase2WeightsKernels.cu

```diff
@@ -0,0 +1,62 @@
+#include <cuda.h>
+
+#include "FWCore/Utilities/interface/CMSUnrollLoop.h"
+#include "DataFormats/EcalRecHit/interface/EcalUncalibratedRecHit.h"
+#include "DataFormats/EcalDigi/interface/EcalLiteDTUSample.h"
+#include "DataFormats/EcalDigi/interface/EcalConstants.h"
+
+#include "EcalUncalibRecHitPhase2WeightsKernels.h"
+
+namespace ecal {
+  namespace weights {
+
```

Comment:
A general comment about kernels: can you rewrite the loops so that the kernel can work with any number of threads and blocks ?
The general approach could be something like
```c++
// the part that copies data from global to shared memory goes here

auto first = threadIdx.x + blockIdx.x * blockDim.x;
auto stride = blockDim.x * gridDim.x;
for (int tx = first; tx < nchannels; tx += stride) {
   // the part that does the computation for each channel goes here
}
```

Reply:
By the way, you prefer to introduce this update to the kernel logic in a future PR, rather than holding up this one, that's OK by me.
In this case, please
  - revert these changes to the kernel
  - add it to the TO DO list in the same issue as https://github.com/cms-sw/cmssw/pull/37695/commits/50eca144f8030043955f2540e972e2c8437608f4#r911002438